File: ProjectCS/OutlinedTextTableRenderer.cs
using System;
using System.IO;
using System.Text;

namespace SessionResultsCS
{
    public class OutlinedTextTableRenderer : StyledTextTableRenderer
    {
        public new static OutlinedTextTableRenderer Default { get; } = new OutlinedTextTableRenderer('+', '-', '|');


        public OutlinedTextTableRenderer(char crossSymbol, char horizontalSymbol, char verticalSymbol) :
            base(crossSymbol, horizontalSymbol, verticalSymbol) { }

        public override string RenderTable(GroupTableData data)
        {
            string table = base.RenderTable(data);
            int width = table.IndexOf(Environment.NewLine);
            string outline = CrossSymbol + new string(HorizontalSymbol, width) + CrossSymbol;

            StringBuilder builder = new StringBuilder();

            builder.AppendLine(outline);
            using (StringReader reader = new StringReader(table))
            {
                string? line;
                while ((line = reader.ReadLine()) != null) {
                    builder.Append(VerticalSymbol);
                    builder.Append(line);
                    builder.Append(VerticalSymbol);
                    builder.AppendLine();
                }
            }
            builder.Append(outline);

            return builder.ToString();
        }
    }
}

File: Project/GroupTable/OutlinedTextTableRenderer.cpp
#include "GroupTable/OutlinedTextTableRenderer.hpp"

#include <sstream>

OutlinedTextTableRenderer::OutlinedTextTableRenderer(char crossSymbol, char horizontalSymbol, char verticalSymbol) :
	StyledTextTableRenderer(crossSymbol, horizontalSymbol, verticalSymbol) {}

OutlinedTextTableRenderer::OutlinedTextTableRenderer(const OutlinedTextTableRenderer& other) :
	StyledTextTableRenderer(other) {}

std::string OutlinedTextTableRenderer::RenderTable(const GroupTableData& data) {
    std::string table = StyledTextTableRenderer::RenderTable(data);
    int width = table.find('\n');
    std::string outline = m_crossSymbol + std::string(width, m_horizontalSymbol) + m_crossSymbol;

    std::stringstream resultss;

    std::stringstream tabless(table);
    std::string buf;
    resultss << outline << std::endl;
    while (std::getline(tabless, buf)) {
        resultss << m_verticalSymbol << buf << m_verticalSymbol << std::endl;
    }
    resultss << outline;

    return resultss.str();
}

std::unique_ptr<OutlinedTextTableRenderer> OutlinedTextTableRenderer::GetDefault() {
    return std::make_unique<OutlinedTextTableRenderer>('+', '-', '|');
}
